src/common-ssh/user.c

```diff
@@ -80,3 +82,15 @@ int guac_common_ssh_user_import_key(guac_common_ssh_user* user,
 
 }
 
+int guac_common_ssh_user_import_public_key(guac_common_ssh_user* user,
+        char* public_key) {
+
+    /* Free existing public key, if present */
+    free(user->public_key);
```

Comment:
Should this perhaps use the same logic as the above function, with a check to see if the key is null or not and then a call to `guac_common_ssh_key_free()`?

Reply:
I skipped the null check because free in C is null pointer safe. I can still add a null check though, it shouldn't do any harm but thought that's not required. 
User->private_key is of type guac_common_ssh_key which includes both key and the optional passphrase so we call guac_common_ssh_key_free() to free the key whereas user->public_key is a simple char* type.

Reply:
Okay, then it's probably fine as-is.